cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1153,13 +1149,15 @@ public Secret generateCertificatesSecret(ClusterCa clusterCa, ClientsCa clientsC
     /* test */ List<ContainerPort> getContainerPortList(KafkaPool pool) {
         List<ContainerPort> ports = new ArrayList<>(listeners.size() + 3);
 
-        if (!useKRaft || pool.isController()) {
+        if ((pool.isBroker() && !pool.isController() && kafkaMetadataConfigState.isZooKeeperOrMigration()) || pool.isController()) {
             // The control plane listener is on all nodes in ZooKeeper based clusters and on nodes with controller role in KRaft
+            // this excludes all the KRaft broker-only nodes even during the migration
             ports.add(ContainerUtils.createContainerPort(CONTROLPLANE_PORT_NAME, CONTROLPLANE_PORT));
         }
 
         // Replication and user-configured listeners are only on nodes with the broker role (this includes all nodes in ZooKeeper based clusters)
-        if (pool.isBroker()) {
+        // or controllers during the migration because they need to contact brokers
```

Comment:
You are opening port on them -> so I guess they need to be contacted by brokers and not contact brokers?

```suggestion
        // or controllers during the migration because they need to be contacted by brokers?
```